docs/deprecation_policy.md

```diff
@@ -0,0 +1,167 @@
+---
+assignees:
+- bgrant0607
+- lavalamp
+- thockin
+
+---
+
+# Kubernetes Deprecation Policy
+
+Kubernetes is a large system with many components and many contributors.  As
+with any such software, the feature set naturally evolves over time.  Sometimes
+we realize that something we have done in the past, whether that is some API, a
+flag, or even a whole feature, needs to be replaced or removed.  Kubernetes is
+committed to not breaking users, so we can’t just make breaking changes.  This
+is when deprecation is triggered.
+
+This document details the policies we hold on deprecation for various facets of
+the system.
+
+## Deprecating parts of the API
+
+Since Kubernetes is an API-driven system, it’s not surprising that the API
+would evolve as our understanding of the problem space evolves.  The Kubernetes
+API is actually a set of APIs, called “API groups”, and each API group is
+versioned independently.  [API versions](http://kubernetes.io/docs/api/)
+fall into 3 main tracks, each of which has different policies for deprecation:
+
+| Example  | Track                            |
+|----------|----------------------------------|
+| v1       | GA (generally available, stable) |
+| v1beta1  | Beta (pre-release)               |
+| v1alpha1 | Alpha (experimental)             |
+
+A given release of Kubernetes can support any number of API groups and any
+number of versions of each.
+
+The following rules govern the deprecation of elements of the API.  This
+includes:
+   * REST resources (aka API objects)
+   * Fields of REST resources
+   * Enumerated or constant values
+These rules are enforced between official releases, not between
+arbitrary commits to master or release branches.
+
+Rule #1: API elements may only be removed by incrementing the version of the
+API group.
+
+Once an API element has been added to an API group at a particular version, it
```

Comment:
once we publish an alpha API, I think it's unreasonable to change it in incompatible ways.  Revving APIs should be cheap (I know it's not but @lavalamp is on it :)